esp-hal/src/i2c/master/mod.rs

```diff
@@ -92,6 +92,9 @@ const MAX_ITERATIONS: u32 = 1_000_000;
 pub enum Error {
     /// The transmission exceeded the FIFO size.
     ExceedingFifo,
+    #[cfg(any(esp32, esp32s2))]
+    /// Read limit of 32 bytes or transfer limit of 31 bytes exceeded.
+    ExceedingTransactionSize,
```

Comment:
Why is this different from `ExceedingFifo`?

Reply:
I was not sure if `ExceedingFifo` is descriptive enough (?) for this error...
I'm still not sure though, so if you think we can re-use already existing `ExceedingFifo` - I'll do the fix

Reply:
`ExceedingTransactionSize` doesn't immediately make sense to me, as I'm not sure what exactly had been exceeded.

`ExceedingTransactionSizeLimit` is probably what you want.

Reply:
> I was not sure if ExceedingFifo is descriptive enough (?) for this error...

It shouldn't ever happen, otherwise we have a bug. The HAL is responsible for papering over length issues if possible. (This also means that a panic is actually not a bad tool here - it's a bug to hit it).

I'd rather not introduce a new error variant for something that we shouldn't let happen in the first place. But trying to read/write longer data than what fits into the FIFO is IMHO pretty much covered by `ExceedingFifo`. If that isn't descriptive enough, then it should be made more descriptive. Note that we already return that error in `setup_write` or `setup_read` for a very similar problem.